torch/nn/parallel/distributed.py

```diff
@@ -1542,6 +1535,17 @@ def forward(self, *inputs, **kwargs):
         self._clear_grad_buffer()
         return output
 
+    def forward(self, *inputs, **kwargs):
+        with torch.autograd.profiler.record_function("DistributedDataParallel.forward"):
+            if self._delay_all_reduce_all_params:
+                output = self.module.forward(*inputs, **kwargs)
+                self._clear_grad_buffer()
```

Comment:
why is this needed only for delay AR?

Reply:
Thanks for pointing out. I think I can move `self._clear_grad_buffer()` to `_post_forward()` as well. Let me update.
PS: The motivation is to move all operations before `self.module.forward()` into `_pre_forward()`, and operations after `self.module.forward()` into `_post_forward()`. So that `_pre_forward()` and `_post_forward()` can be reused in composable APIs.